File: send-data-from-one-fragment-to-another-java-version/app/src/main/java/com/example/senddatafromonefragmmenttoanotherjavaversion/FragmentTwo.java
package com.example.senddatafromonefragmmenttoanotherjavaversion;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class FragmentTwo extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.activity_fragment_two, container, false);
        String inputText = getArguments().getString("inputText");

        TextView textView = view.findViewById(R.id.outPutTextView);
        textView.setText(inputText);
        return view;
    }
}

File: send-data-from-one-fragment-to-another-fragment-using-kotlin/app/src/main/java/com/example/senddatafromonefragmenttoanotherusingkotlin/FragmentTwo.kt
package com.example.senddatafromonefragmenttoanotherusingkotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.activity_fragment_two.view.*

class FragmentTwo : Fragment() {
    var inputText: String? = ""

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val rootView = inflater.inflate(R.layout.activity_fragment_two, container, false)
        inputText = arguments?.getString("inputText")
        rootView.outPutTextView.text = inputText
        return rootView
    }
}